Source/bmalloc/bmalloc/bmalloc.h

```diff
@@ -62,9 +63,16 @@ inline pas_primitive_heap_ref& heapForKind(Gigacage::Kind kind)
 BINLINE void* tryMalloc(size_t size, CompactAllocationMode mode, HeapKind kind = HeapKind::Primary)
 {
 #if BUSE(LIBPAS)
-    if (!isGigacage(kind))
-        return bmalloc_try_allocate_inline(size, asPasAllocationMode(mode));
-    return bmalloc_try_allocate_auxiliary_inline(&heapForKind(gigacageKind(kind)), size, asPasAllocationMode(mode));
+    if (!isGigacage(kind)) {
+        void* allocation = bmalloc_try_allocate_inline(size, asPasAllocationMode(mode));
+        if (mode == CompactAllocationMode::NonCompact && MAR_isAddressInQualifyingPage(allocation))
+            MAR_didAllocate(&marRegistry, allocation, size);
+        return allocation;
```

Comment:
Doesn't this form become non-trail call? The important thing is that we must not call something in non tail call for the slow path to ensure that the function is kept as a leaf function.